genesyscloud/oauth_client/resource_genesyscloud_oauth_client.go

```diff
@@ -49,15 +49,21 @@ func createOAuthClient(ctx context.Context, d *schema.ResourceData, meta interfa
 	state := d.Get("state").(string)
 
 	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
-	oauthClientProxy := getOAuthClientProxy(sdkConfig)
+	oauthClientProxy := GetOAuthClientProxy(sdkConfig)
 
 	roles, diagErr := buildOAuthRoles(d)
 	if diagErr != nil {
 		return diagErr
 	}
 
+	//Before we create the oauth client take any roles on it and update the user
+	diagErr = updateTerraformUserWithRole(ctx, sdkConfig, roles)
```

Comment:
This is a new function.  Basically, if we create an oauth client we need to make sure any roles that were assigned to the oauth client are assigned to the user running the process.  This process will only work if the user is running user-based access token (e.g. like in CX accelerators.). If the user is running under an oauth credential and they do not have the role assigned to the oauth client, the code will fail with an error.